From: Damjan Jovanovic <damjan.jov@gmail.com>
Subject: [PATCH v3 7/7] wldap32: supply our username as SASL_CB_AUTHNAME too
Message-Id: <CAJm2B-=uzRw3aH5Q+j6hkUK_CJXPY_UOA7eTiiJ8oL84iVjssQ@mail.gmail.com>
Date: Thu, 18 Feb 2021 18:50:02 +0200

Try 3 gives up the attempt to provide credentials in an
authentication method specific form, and just provides our
username as the authentication username (SASL_CB_AUTHNAME) too.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=50572
Signed-off-by: Damjan Jovanovic <damjan.jov@gmail.com>
---
 dlls/wldap32/libldap.h | 1 +
 dlls/wldap32/bind.c    | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/wldap32/libldap.c b/dlls/wldap32/libldap.c
index 1498dc49fe6..fa18b9597a7 100644
--- a/dlls/wldap32/libldap.c
+++ b/dlls/wldap32/libldap.c
@@ -198,7 +198,7 @@ static int sasl_interact( LDAP *ld, unsigned flags, void *defaults, void *intera
             sasl->result = id->Domain;
             sasl->len = id->DomainLength;
         }
-        else if (sasl->id == SASL_CB_USER)
+        else if (sasl->id == SASL_CB_AUTHNAME || sasl->id == SASL_CB_USER)
         {
             sasl->result = id->User;
             sasl->len = id->UserLength;

